providers/sendgrid/src/lib/sendgrid.provider.ts

```diff
@@ -41,7 +41,7 @@ export class SendgridEmailProvider implements IEmailProvider {
     });
 
     return {
-      id: response[0]?.headers['x-message-id'],
+      id: options.id || response[0]?.headers['x-message-id'],
```

Comment:
Why not use the 'x-message-id' here regardless? We always can get our message If needed

Reply:
Because in Sendgrid the message _id will be the identifier in some other it will be set by the provider and needs to be returned. x-message-id is not sent in the webhook later so we can not indentify it by it.